pkgs/development/libraries/rocsparse/default.nix

```diff
@@ -0,0 +1,148 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, rocm-cmake
+, rocm-runtime
+, rocm-device-libs
+, rocm-comgr
+, rocprim
+, hip
+, gfortran
+, git
+, fetchzip ? null
+, gtest ? null
+, boost ? null
+, python3Packages ? null
+, buildTests ? false
+, buildBenchmarks ? false # Seems to depend on tests
+}:
+
+assert (buildTests || buildBenchmarks) -> fetchzip != null;
+assert (buildTests || buildBenchmarks) -> gtest != null;
+assert (buildTests || buildBenchmarks) -> boost != null;
+assert (buildTests || buildBenchmarks) -> python3Packages != null;
+
+let
+  matrices = lib.optionalAttrs (buildTests || buildBenchmarks) import ./deps.nix {
+    inherit fetchzip;
+    mirror1 = "https://sparse.tamu.edu/MM";
+    mirror2 = "https://www.cise.ufl.edu/research/sparse/MM";
+  };
+in stdenv.mkDerivation rec {
+  pname = "rocsparse";
+  rocmVersion = "5.3.1";
+  version = "2.3.2-${rocmVersion}";
+
+  outputs = [
+    "out"
+  ] ++ lib.optionals (buildTests || buildBenchmarks) [
+    "test"
```

Comment:
How can I run the tests? I get
```bash
$ result-test/bin/rocsparse-test
"hipErrorNoBinaryForGpu: Unable to find code object for all current devices!"
fish: Job 1, 'result-test/bin/rocsparse-test' terminated by signal SIGABRT (Abort)
```
I guess it either needs something in my environment or I have to specify the GPU somewhere?

Reply:
Probably yes, see: https://github.com/Madouura/nixos-config/blob/0e4236c18a468abd300b3df5c2657dba6f763ec0/hosts/machines/ura/modules/system/hardware.nix#L6-L9